ddtrace/internal/ci_visibility/git_client.py

```diff
@@ -111,14 +114,18 @@ def _get_repository_url(cls, tags={}, cwd=None):
 
     @classmethod
     def _get_latest_commits(cls, cwd=None):
-        # type: (Optional[str]) -> list[str]
+        # type: (Optional[str]) -> List[str]
         return extract_latest_commits(cwd=cwd)
 
     @classmethod
     def _search_commits(cls, requests_mode, base_url, repo_url, latest_commits, serializer, _response):
-        # type: (int, str, str, list[str], CIVisibilityGitClientSerializerV1, Optional[Response]) -> list[str]
+        # type: (int, str, str, List[str], CIVisibilityGitClientSerializerV1, Optional[Response]) -> Optional[List[str]]
         payload = serializer.search_commits_encode(repo_url, latest_commits)
         response = _response or cls._do_request(requests_mode, base_url, "/search_commits", payload, serializer)
+        if response.status >= 400:
+            log.warning("Response status: %s", response.status)
+            log.warning("Response body: %s", response.body)
+            return None
```

Comment:
Since we're not really using the fact that the return value is `None` in `_run_protocol()`, maybe returning an empty list is more appropriate.